docs/technical/security.md

```diff
@@ -0,0 +1,201 @@
+
+### Containers
+
+Docker Containers most commonly use base images which contain various component which themselves may have security vulnerabilities.  Known CVEs may exist for components within these images which may expose vulnerabilities aside from the code and dependencies directly shipped by this project.
+
+Tools exist to scan containers for vulnerabilities.  One such tool is, Container Scan which can be run as part of the CI pipeline.
+
+[Container Scan](https://github.com/marketplace/actions/container-image-scan)
+
+Other container scanning technologies exist if this is found to be unsuitable.
+
+Some consideration should be given to containers which are in production when vulnerabilities are discovered.  This may mean rescanning containers on a regular basis and raising alerts if security issues are discovered.
+
+## Kubernetes
+
+### Keep up with released versions of Kubernetes
+
+AKS regularly make available updated versions of Kubernetes.  These contain new features, but most importantly for this document, security patches.
+
+Whilst in operation - these updates will need to be evaluated an applied.
+
+Other Kubernetes implementations will also need to be regularly patched and maintained.
+
+### Don't run containers as root
+
+Best practice to avoid 'Container Escape' is to not run the container from a root user account.
+
+For the purposes of this project - that would mean creating a low privileged user.
+
+```Dockerfile
+...
+RUN adduser \
+  --disabled-password \
+  --home /app \
+  --gecos '' app \
+  && chown -R app /app
+USER app
+...
+```
+
+### Run in a non-default namespace
+
+Kubernetes Namespaces are separate from one another although they can communicate with services in other namespaces.   Namespaces offer a way to partition services and make them easier to manage.  
+
+RBAC can be applied at the namespace level and will give admins more granularity and control over the services deployed to namespaces under their control.
+
+For the purposes of this project - DMR and CentOps should reside in their own namespaces ('byk-dmr' and 'byk-centops', for instance) .  Hosted bots *could* reside in a shared 'bots' namespace as they may have similar access patters and permissions (unless hosted bots have special security considerations)  
+
+### Specify memory and CPU quotas
+
+In order to prevent containers affecting other pods in the cluster it's often recommended to specify CPU usage and memory limits.  This can be done in the helm chart.
+
+```yaml
+...
+resources:
+  limits:
+    cpu: 100m
+    memory: 128Mi
+  requests:
+    cpu: 100m
+    memory: 128Mi
+...
+```
+
+The process of determining sensible limits can be a tricky one, however.  K8s will terminate the running pod which has gone beyond it's limits, so these values should be carefully chosen.  In the event of a pod exceeding this values - K8s will recycle the pod, so sensible values are a must.

```

Comment:
And there is at least one more thing to consider when setting requests and limits. The autoscaling. In order to scale pods there must be difference between limits and requests.

Reply:
Quite right - added a sentence to mention this.